teps/0017-results-api.md

```diff
@@ -0,0 +1,711 @@
+representing meta-configs of TaskRuns (i.e. DSLs), or other execution-like data
+that is not handled by Tekton directly.
+
+#### Example
+
+```proto
+results: {
+  name: "namespace/default/results/sample-tekton-result"
+  created_time: { seconds: 1234 }
+
+  annotations: {
+    "env": "ci"
+  }
+
+  executions: {
+    pipeline_run: {
+      # ...
+    }
+  }
+  executions: {
+    task_run: {
+      # ...
+    }
+  }
+  executions: {
+    task_run: {
+      # ...
+    }
+  }
+
+  # These aren't real types (yet), just examples of the kind of data we could expect.
+  extensions: {
+    key: "tekton-chains"
+    value: {
+      type_url: "tekton.chains.v1.SHA"
+      value: "aHVudGVyMg=="
+    }
+  }
+  extensions: {
+    key: "notifications"
+    value: {
+      type_url: "tekton.notifications.v1.Notification"
+      value: "{type: \"cloudevent\", url: "https://example.com", status: 200}"
+    }
+  }
+}
+```
+
+Executions serve as an abstraction over runtime information. They may contain:
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Making Results part of the core Pipeline controller
+
+While we expect that this will be a commonly used component for many
+installations, we do not want to make this a requirement for every installation.
+Users should be able to enable / disable this component freely to fit their
+needs.
+
+## Infrastructure Needed (optional)
+
+<!--
+Use this section if you need things from the project/SIG.  Examples include a
+new subproject, repos requested, github details.  Listing these here allows a
+SIG to get the process for these resources started right away.
+-->
+
+- New repository: github.com/tektoncd/results
+
+## Upgrade & Migration Strategy (optional)
+
+<!--
+Use this section to detail wether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
+
+We be consistent with the Pipelines
+[API compatibility policy](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md).
```

Comment:
I'd state clearly this means that while Results is in alpha you can make any breaking changes you want.

Reply:
Done.